tools/arriba.wdl

```diff
@@ -0,0 +1,374 @@
+                'intronic',
+                'in_vitro',
+                'intragenic_exonic',
+                'internal_tandem_duplication'
+            ]
+        }
+        feature_name: {
+            description: "List of feature names to use in GTF."
+            help: "The Arriba default it designed to handle RefSeq, GENCODE, or ENSEMBL format annotations. `feature_name` expects a string of space separated options. The required fields are `gene_name`, `gene_id`, `transcript_id`, `feature_exon`, and `feature_CDS`. The fields should space separated. The values should be provided with `field=value`. Mutliple values can be provided and separated by a pipe (`|`), e.g. `=value1|value2`. A complete example is `gene_name=gene_name|gene_id gene_id=gene_id transcript_id=transcript_id feature_exon=exon feature_CDS=CDS`."
+            external_help: "https://arriba.readthedocs.io/en/latest/command-line-options/"
+            common: false
+        }
+        strandedness: {
+            description: "Strandedness of the input data."
+            external_help: "https://arriba.readthedocs.io/en/latest/command-line-options/"
+            choices: [
+                "auto",
+                "yes",
+                "no",
+                "reverse"
+            ]
+        }
+        mark_duplicates: "Mark duplicates in the input BAM file with Arriba."
+        report_additional_columns: "Report additional columns ['fusion_transcript', 'peptide_sequence', 'read_identifiers'] in the discarded fusions file."
+        fill_gaps: "Fill gaps in assembled transcripts with reference bases. Expands the fusion sequence to the complete sequence of the fusion gene."
+        max_e_value: "Maximum E-value for read support."
+        max_mismappers: "Maximum fraction of mismapped reads in the fusion region."
+        max_homolog_identity: "Maximum fraction of homologous sequence for genes."
+        max_kmer_content: "Maximum fraction of repetitive 3-mer content in the fusion region."
+        max_mismatch_pvalue: "Maximum p-value for mismatches in the fusion region."
+        quantile: "Genes with expression above the given quantile are eligible for filtering."
+        exonic_fraction: "Minimum fraction of exonic sequence between breakpoints."
+        coverage_fraction: "Minimum fraction of viral contig transcription."
+        min_itd_allele_fraction: "Minimum supporting read fraction for internal tandem duplications."
+        max_genomic_breakpoint_distance: "With 'wgs_svs', threshold for relating genomic and transcriptomic events."
+        min_supporting_reads: "Minimum number of supporting reads for a fusion."
+        homopolymer_length: "Maximum homopolymer length adjacent to breakpoints."
+        read_through_distance: "Minimum distance between breakpoints for read-through events."
+        min_anchor_length: "Minimum anchor length for split reads."
+        many_spliced_events: "Recover fusions with at least this many spliced breakpoints."
```

Comment:
Should we do something explicit like `common: false`? Not that different to what we're doign now, but maybe makes it clearer that these probably shouldn't be changed.

Reply:
I thought about doing `common: false`, but that is potentially confusing. There would be 3 cases, `common: true`, `common: false`, and the most frequent case where there is no `common` entry. To me, it only makes sense that `null` case should be considered equivalent with the `false` case. I don't want to get fall into the `T/F/Null` trap.

I think it's the same with the `deprecated` key we introduced. In the absence of a `deprecated` key, it's safe to assume `deprecated: false`.

Reply:
To me, the absence of `common: true` is an indication that most people shouldn't need to mess with the parameters. They're also sorted to the bottom of the list. I'm concerned that if we move toward a community resource model, where we need to expose all tool parameters, we're going to end up with a ton of these types of situations. It's a lot to read and maintain as well as requiring some consensus on what `advanced` would be.

Reply:
Let's view this through the lens of `sprocket doc`.

* `common: true` [will] mean put this right under the required parameters and highlight it
* `common: false || NULL` means no special treatment

Are those two cases enough?
Going off memory here, but our current convention is to only start labelling params as `common` once a task has >10-or-so parameters. Should we revisit that and start labelling `common` more frequently?

Reply:
I'm fine with only using `common` for now. It's easy enough to add other labels later if we see the need. But if we're going to keep with the one-label/two-conditions situation, I'd like to see us more consistently use the `true` condition. Doesn't have to hold up this PR, but something for us to keep in mind for the future.

Reply:
I just thought of this, but what if we added an additional level/object for `sprocket`?
```
        chimSegmentMin: {
            description: "minimum length of chimeric segment length, if ==0, no chimeric output",
            tool_default: 0,
            sprocket: {
                 category: common
            }
        }
```
I could immediately see `common` and `advanced` for `category` choices. Are there other things we might want to explicitly write for `sprocket`?

Reply:
From Slack, we decided to continue with just `common: true` for now. We may refactor to 
```
 sprocket: {
    category: common
}
```
in the future. This would be a controlled vocabulary. We would also want additional use cases for the `sprocket` object before proceeding.